### DISCUSSION TITLE
Windows Containers?

### DISCUSSION THREAD

**Discussion by sebjsmith** [dorssel/usbipd-win#732] [Q&A]
I am trying to install this in a Windows Container but am getting an error running msiexec. I was trying to bundle this software into a container along with some Windows only software to connect to the USB device.

I think it fails here and then rolls back:

`MSI (s) (34:2C) [18:42:08:352]: Hello, I'm your 64bit Elevated Non-remapped custom action server.
ExecFirewallExceptions:  Installing firewall exception2 usbipd (C:\Program Files\usbipd-win\usbipd.exe)
ExecFirewallExceptions:  Error 0x800706d9: failed trying to find existing app
ExecFirewallExceptions:  Error 0x800706d9: failed to add/update application exception for name 'usbipd', file 'C:\Program Files\usbipd-win\usbipd.exe'`

I've also come across this post here which leads me to believe this isn't possible.
https://forums.docker.com/t/is-it-possible-to-use-usb-ip-in-windows-container/136490/2

Before I give up and start thinking of different solutions I just wanted to ask here. Thank-you

**Comment by dorssel** [comment]
You want to run the USBIP *server* inside a container? Is that even possible? Can the container access/change host device drivers at all?

If yes: the problem is that the container has no access to Windows Firewall settings. The installer wants to add an exception here: https://github.com/dorssel/usbipd-win/blob/master/Installer/Server.wxs#L20-L28. This could be made to fail silently by adding IgnoreFailure (see https://wixtoolset.org/docs/schema/firewall/firewallexception/). But I would have to first assess the possible drawbacks. For the time being, you could build your own installer.

**Reply by sebjsmith** [reply]
I'm hoping to run the USBIP client only in the container.

I will see how far I get with IgnoreFailure on the firewall settings.

Many thanks for your response.

**Reply by dorssel** [reply]
> I'm hoping to run the USBIP client only in the container.

Ah, that is what I thought. You do realize this project is only server side? There is no reason to install `usbipd-win` at all. What you need is a client, which this project does not contain. Look through older issues/discussions on how some people made client-side Windows work using other tools.

**Reply by jojoatt** [reply]
> I'm hoping to run the USBIP client only in the container.
> 
> 
> 
> I will see how far I get with IgnoreFailure on the firewall settings.
> 
> 
> 
> Many thanks for your response.

Hello, did you manage to run an USB/IP client on windows container ? I'm trying to do it but it doesn't work and it don't see to be possible with windows containers
